src/test/java/com/fasterxml/jackson/core/write/GeneratorCopyTest.java

```diff
@@ -100,4 +100,38 @@ void copyObjectTokens()
         gen.close();
         assertEquals("{\"a\":1}", sw.toString());
     }
+
+    @Test
+    void copyNumericTokensExactly()
+            throws IOException
+    {
+        JsonFactory jf = JSON_F;
+        final String DOC = "{ \"a\":0.123456789123456789123456789, \"b\":[" +
+            "{ \"c\" : null, \"d\" : 0.123456789123456789123456789 }] }";
+        JsonParser jp = jf.createParser(new StringReader(DOC));
+        StringWriter sw = new StringWriter();
+        JsonGenerator gen = jf.createGenerator(sw);
+
+        assertToken(JsonToken.START_OBJECT, jp.nextToken());
+        gen.copyCurrentStructureExact(jp);
+        // which will advance parser to matching end Object
+        assertToken(JsonToken.END_OBJECT, jp.currentToken());
+        jp.close();
```

Comment:
with new tests, I prefer to see try with resources used - https://www.baeldung.com/java-try-with-resources